src/internal/connector/sharepoint/data_collections.go

```diff
@@ -108,25 +120,30 @@ func collectLists(
 	tenantID, siteID string,
 	updater statusUpdater,
 	ctrlOpts control.Options,
+	errs *fault.Errors,
 ) ([]data.BackupCollection, error) {
 	logger.Ctx(ctx).With("site", siteID).Debug("Creating SharePoint List Collections")
 
 	spcs := make([]data.BackupCollection, 0)
 
-	tuples, err := preFetchLists(ctx, serv, siteID)
+	lists, err := preFetchLists(ctx, serv, siteID, errs)
 	if err != nil {
 		return nil, err
 	}
 
-	for _, tuple := range tuples {
+	for _, tuple := range lists {
+		if errs.Err() != nil {
+			break
+		}
+
 		dir, err := path.Builder{}.Append(tuple.name).
 			ToDataLayerSharePointPath(
 				tenantID,
 				siteID,
 				path.ListsCategory,
 				false)
 		if err != nil {
-			return nil, errors.Wrapf(err, "failed to create collection path for site: %s", siteID)
+			errs.Add(clues.Wrap(err, "creating list collection path").WithClues(ctx))
```

Comment:
Should we retain info about `siteID`? Unless I missed this getting added anywhere else.

Reply:
Since siteID is equal to the resource owner, that info is also already present in the context.